InformationScripting/src/queries/QueryExecutor.cpp

```diff
@@ -57,11 +67,17 @@ Interaction::CommandResult* QueryExecutor::execute()
 		}
 		else
 		{
-			return new Interaction::CommandResult(new Interaction::CommandError(results[0].errors()[0]));
+			hasError = true;
+			errorMessage = results[0].errors()[0];
 		}
 	}
 
-	return new Interaction::CommandResult();
+	if (queries_.empty()) delete this;
```

Comment:
I still feel a bit uneasy about this. Could you quickly have a look at Qt functionality for late deleting stuff? I remember that they had some mechanism to delete stuff later (maybe schedule the deletion in an event?) I guess we could this this ourselves, regardless of QT support, just post an event that deletes.